### DISCUSSION TITLE
Bnd_OBB::GetVertex

### DISCUSSION THREAD

**Discussion by PVTSoftware** [Macad3D/Macad3D#78] [General]
Hi 
I would to use the Bnd_OBB's method GetVertex . the method accepts a **Pnt** type parameter then the native method accepts an array 
**gp_Pnt theP[8]**. I would like to know if this is an error and how to correct it.

Regards

**Comment by dipts** [comment]
This is a limitation of the WrapperGenerator. Only the ABI type is used there, and that is `gp_Pnt*`. However, CastXml extracts the additional information that the type was originally declared as a fixed size array. One possibility would therefore be to extend the WrapperGenerator in order to evaluate this additional information. To do this, the parser would have to read this information from the CastXML model and save it in the internal model. The ResolveTypesPass could then use this information to change the wrapper type to a (new) ArrayWrapper (which creates the corresponding wrapper code).

Alternatively, an overload can also be created manually for this one function. For this purpose, a define is inserted in each wrapped class, which makes it possible to inject additional code into the declaration. You can find examples of this in the [Macad.Occt/Extensions](https://github.com/Macad3D/Macad3D/tree/main/Source/Macad.Occt/Extensions) directory. If you go this way, remember to enter the additional header file in the [PCH](https://github.com/Macad3D/Macad3D/blob/main/Source/Macad.Occt/OcctPCH.h).

**Reply by PVTSoftware** [reply]
Thanks
